synthesizer/src/process/trace/mod.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod inclusion;
+pub use inclusion::*;
+
+mod query;
+pub use query::*;
+
+use crate::{
+    block::{Execution, Fee, Input, Transition},
+    snark::{Proof, ProvingKey, VerifyingKey},
+    store::BlockStorage,
+};
+use circuit::Assignment;
+use console::{
+    network::prelude::*,
+    program::{InputID, Locator},
+};
+
+use once_cell::sync::OnceCell;
+use std::collections::HashMap;
+
+#[derive(Clone, Debug, Default)]
+pub struct Trace<N: Network> {
+    /// The list of transitions.
+    transitions: Vec<Transition<N>>,
+    /// A map of locators to (proving key, assignments) pairs.
+    transition_tasks: HashMap<Locator<N>, (ProvingKey<N>, Vec<Assignment<N::Field>>)>,
+    /// A tracker for all inclusion tasks.
+    inclusion_tasks: Inclusion<N>,
+
+    /// A tracker for the inclusion assignments.
+    inclusion_assignments: OnceCell<Vec<InclusionAssignment<N>>>,
+    /// A tracker for the global state root.
+    global_state_root: OnceCell<N::StateRoot>,
+}
+
+impl<N: Network> Trace<N> {
+    /// Initializes a new trace.
+    pub fn new() -> Self {
+        Self {
+            transitions: Vec::new(),
+            transition_tasks: HashMap::new(),
+            inclusion_tasks: Inclusion::new(),
+            inclusion_assignments: OnceCell::new(),
+            global_state_root: OnceCell::new(),
+        }
```

Comment:
```suggestion
        Self::default()
```

Reply:
It isn't able to be auto-derived due to `N`